### PR TITLE
net-vpn/i2p: bump to 0.9.44

### PR DESCRIPTION
**Pull request by tharvik**
* simply `cp` works
* dropping old version
* keeping the migration system for now, until 0.9.42 is also dropped

### CODE REVIEW

**Comment by juippis** [comment]
```
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:5: error: package org.mockito does not exist
    [javac] import static org.mockito.Mockito.*;
    [javac]                          ^
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:22: error: package org.mockito does not exist
    [javac] import org.mockito.Mock;
    [javac]                   ^
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:23: error: package org.mockito.junit does not exist
    [javac] import org.mockito.junit.MockitoJUnit;
    [javac]                         ^
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:24: error: package org.mockito.junit does not exist
    [javac] import org.mockito.junit.MockitoRule;
    [javac]                         ^
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:44: error: cannot find symbol
    [javac]     public MockitoRule rule = MockitoJUnit.rule();
    [javac]            ^
    [javac]   symbol:   class MockitoRule
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:46: error: cannot find symbol
    [javac]     @Mock private I2PSocketManager mgr;
    [javac]      ^
    [javac]   symbol:   class Mock
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:47: error: cannot find symbol
    [javac]     @Mock private I2PSocketOptions opts;
    [javac]      ^
    [javac]   symbol:   class Mock
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:48: error: cannot find symbol
    [javac]     @Mock private I2PSocket sock;
    [javac]      ^
    [javac]   symbol:   class Mock
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:44: error: cannot find symbol
    [javac]     public MockitoRule rule = MockitoJUnit.rule();
    [javac]                               ^
    [javac]   symbol:   variable MockitoJUnit
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:74: error: method any in class Matchers cannot be applied to given types;
    [javac]         when(mgr.buildOptions((Properties) any())).thenReturn(opts);
    [javac]                                            ^
    [javac]   required: Class<T>
    [javac]   found: no arguments
    [javac]   reason: cannot infer type-variable(s) T
    [javac]     (actual and formal argument lists differ in length)
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in method <T>any(Class<T>)
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:74: error: cannot find symbol
    [javac]         when(mgr.buildOptions((Properties) any())).thenReturn(opts);
    [javac]         ^
    [javac]   symbol:   method when(I2PSocketOptions)
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:75: error: method any in class Matchers cannot be applied to given types;
    [javac]         when(mgr.connect((Destination) any(), eq(opts))).thenReturn(sock);
    [javac]                                        ^
    [javac]   required: Class<T>
    [javac]   found: no arguments
    [javac]   reason: cannot infer type-variable(s) T
    [javac]     (actual and formal argument lists differ in length)
    [javac]   where T is a type-variable:
    [javac]     T extends Object declared in method <T>any(Class<T>)
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:75: error: cannot find symbol
    [javac]         when(mgr.connect((Destination) any(), eq(opts))).thenReturn(sock);
    [javac]                                               ^
    [javac]   symbol:   method eq(I2PSocketOptions)
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:76: error: cannot find symbol
    [javac]         when(sock.getOutputStream()).thenReturn(out);
    [javac]         ^
    [javac]   symbol:   method when(OutputStream)
    [javac]   location: class I2PSocketEepGetTest
    [javac] /tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:77: error: cannot find symbol
    [javac]         when(sock.getInputStream()).thenReturn(in);
    [javac]         ^
    [javac]   symbol:   method when(InputStream)
    [javac]   location: class I2PSocketEepGetTest
    [javac] 15 errors

BUILD FAILED
/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/build.xml:1853: The following error occurred while executing this line:
/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/build.xml:190: Compile failed; see the compiler error output for details.

Total time: 44 seconds
 * ERROR: net-vpn/i2p-0.9.44::testworld failed (test phase):
 *   eant failed
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called src_test
 *   environment, line 3060:  Called java-pkg-2_src_test
 *   environment, line 1406:  Called eant '-f' 'build.xml' '-Djavac.classpath=//usr/share/junit-4/lib/junit.jar://usr/share/hamcrest-core-1.3/lib/hamcrest-core.jar://usr/share/hamcrest-library-1.3/lib/hamcrest-library.jar' 'junit.test'
 *   environment, line  564:  Called die
 * The specific snippet of code:
 *       ant ${antflags} "${@}" || die "eant failed"
```
Maybe missing some test dep?

**Comment by tharvik** [comment]
> Maybe missing some test dep?

I guess you ran it with `ebuild ... test`, without setting `USE=test` :P

I'm removing the streaming tests in the `src_prepare` as it requires some more recent version than `=dev-java/mockito-1.9.5-r1`. When reenabling theses (and adding mockito to classpath), I get the next build errors.

```
    [javac] Compiling 5 source files to /var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/build/obj
    [javac] /var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:23: error: package org.mockito.junit does not exist
    [javac] import org.mockito.junit.MockitoJUnit;
    [javac]                         ^
    [javac] /var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:24: error: package org.mockito.junit does not exist
    [javac] import org.mockito.junit.MockitoRule;
    [javac]                         ^
    [javac] /var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:44: error: cannot find symbol
    [javac]     public MockitoRule rule = MockitoJUnit.rule();
    [javac]            ^
    [javac]   symbol:   class MockitoRule
    [javac]   location: class I2PSocketEepGetTest
    [javac] /var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/test/junit/net/i2p/client/streaming/I2PSocketEepGetTest.java:44: error: cannot find symbol
    [javac]     public MockitoRule rule = MockitoJUnit.rule();
    [javac]                               ^
    [javac]   symbol:   variable MockitoJUnit
    [javac]   location: class I2PSocketEepGetTest
    [javac] 4 errors

BUILD FAILED
/var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/build.xml:1853: The following error occurred while executing this line:
/var/tmp/portage/net-vpn/i2p-0.9.44/work/i2p-0.9.44/apps/ministreaming/java/build.xml:190: Compile failed; see the compiler error output for details.
```

**Comment by juippis** [comment]
Nope, it's run with `emerge`. All of the test deps are installed...

File: net-vpn/i2p/i2p-0.9.44.ebuild [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+# Until the deps reach other arches
+KEYWORDS="~amd64 ~x86"
+IUSE="nls test"
+RESTRICT="!test? ( test )"
+
+# dev-java/ant-core is automatically added due to java-ant-2.eclass
+CP_DEPEND="
```
Comment (juippis) [inline net-vpn/i2p/i2p-0.9.44.ebuild]: Please use `COMMON_DEPEND` instead.

**Comment by tharvik** [comment]
Okay, finally found it; as I'm sed'ing the `build.xml` after it's rewrite, it might remove the *streaming tests, or not, depending on the order of attributes in the rewritten file. As, to the best of my knowledge, this order is not stable, it works in some runs and fails in others.

To avoid that, I'm moving the sed'ing prior to the rewrite, can you try it again?

**Pull request closed without merging**